Web migration

AndrewHanley opened the pull request.

areyeslo committed:
Remove objects for first nations and municipalities when empty. Nothing to save there.

Commit:
Migrated API CIOT Code

areyeslo committed:
getting nearest in backend instead of db.

areyeslo committed:
Changes in text through CIOT

areyeslo committed:
Change serializers to prefetch results. Add spinner in Dashboard when waiting for list of promoted opportunities. Fix get user queryset.

areyeslo committed:
Changes in text for email

areyeslo committed:
Fix invalid address being able to go to step 2 after finishing request lat/long

Commit:
Migration of CIOT-Web App

jjstratton commented on README.md:
````diff
@@ -1,2 +1,117 @@
+Copy a local config templates:
+
+```
+cp .\ciot-api\.env.template .\ciot-api.env
````
I think the docker compose is missing a  \ before the .env

areyeslo commented on ciot-web/src/components/AddressSearchBar/AddressSearchBar.js:
```diff
@@ -0,0 +1,118 @@
+        placeholder="eg. 635 Humboldt St, Victoria, BC"
+        className=" bcgov-text-input"
+        id="addressSearch"
+        value={currentAddress || ""}
```
The update version in react made the store unstable. It is not updating correctly. For instance this component is getting the value from the state using a selector. However, it is not updating properly. I would suggest to get back to the current version and just get the application running separately. Then, we can update the versions if we have the time and budget.

areyeslo committed:
Avoid triggering useEffect if you do not need to

areyeslo committed:
Disable button with loader while opportunity submission is in progress.

areyeslo committed:
Change to include civic number in text when providing an opportunity address.

areyeslo committed:
Adding total of properties in paginator. Bug fix to show no opportunities when user is deleted.

areyeslo committed:
Add header to Manage Users.

areyeslo committed:
Adding municipality and regional district values.

areyeslo committed:
Reduce distance from 100 km to 150 mts.

Commit:
Updated package.json to copy from CIT repo

areyeslo committed:
Hide Manage Users in menu.

areyeslo committed:
Changes in text for opportunity creation.

areyeslo committed:
Change in reducer

Commit:
Updated opportunity reducer to create copy of state to ensure updates from react-redux don't stop state changes from occurring.

Commit:
Merge branch 'Web-Migration' of https://github.com/bcgov/CIOT into Web-Migration

Commit:
Updated Review to useEffect for Posting Opportunity

areyeslo committed:
Changes in the email text.

Commit:
Updated opportunity approval Redux code

Commit:
Updated environment template

areyeslo committed:
Send put request every time the user clicks the button regardless lastAdmin is not changed.

areyeslo committed:
Clean code no needed. Creating new object at the beginning instead of creating it for each case.

areyeslo committed:
Remove unused code, components used by CIT.

areyeslo committed:
Open View property details in new tab.

areyeslo committed:
Spinner when data is loading.

areyeslo committed:
Change layout in map to satellite.

areyeslo committed:
Add match precision to discard invalid addresses in Geocoder call to retrieve valid addresses.

areyeslo committed:
- Remove NEW and NEW Edited statuses.
- New migration to update opportunities to Pending Review and delete old state.

areyeslo committed:
Notification email changes for opportunity creation.

areyeslo committed:
Update approval status for NCOM

areyeslo committed:
reset opportunity to allow the spinner to run when opportunity id is empty.

areyeslo committed:
Adding FK in indianReserveBandDistance to community table.

areyeslo committed:
Adding FirstNations to serializer. Removing SerializerMethodField.

areyeslo committed:
Creating new serializer for municipalities. Changes in database to connect municipalityDistance to community table to obtain population.

areyeslo committed:
Changes in prefetching.

areyeslo committed:
Reduced queries for opportunities.

areyeslo committed:
Fix flags to change to next page. Remove radio button component. Edit text for opportunities.

areyeslo committed:
Do not delete geometry when modifying parcel size, it is needed in further stage.

areyeslo committed:
Changes in text for dashboard.

areyeslo committed:
Remove Close filter panel

areyeslo committed:
Remove LinesEllipsis and truncate string invocating a function.

areyeslo committed:
Remove Residential zoning from search zoning.

areyeslo committed:
Remove exclude unknown and add it by default with No values in search page.

areyeslo committed:
Remove nearest lake and river taking more than half of the time of query execution.

areyeslo committed:
Hide Nearest River and Nearest Lake in Front End.

areyeslo committed:
Fix Connectivity filter to be included in query.

areyeslo committed:
Add spinner to OpportunityPage to match OpportunityApprovePage

areyeslo committed:
Remove Nearby Resources from the map.

areyeslo committed:
Add contributor and investor disclaimer.

areyeslo committed:
Remove old disclaimer text.

areyeslo committed:
Changes in styling Search

areyeslo committed:
add margin bottom

areyeslo committed:
CIT-807: Remove text on How it works page

areyeslo committed:
CIT-810: Text change under Your Investment Opportunities

areyeslo committed:
CIT-787: Filter panel text edits.

areyeslo committed:
CIT-773: Change email for Contact Us.

areyeslo committed:
Missing space in disclaimer.

areyeslo committed:
Remove disclaimer

areyeslo committed:
Swap disclaimers.

areyeslo committed:
Additional references to use of term on disclaimer contributor.

areyeslo committed:
Added send-update-reminders in commands folder.

areyeslo committed:
Change logic for Search Opportunities. When user selects No,then disable the filter. Otherwise, obtain the records with Yes.

areyeslo committed:
Changes during opportunity creation. Added new disclaimers in proper location.

areyeslo committed:
Removing migrations from cherry-pick and Disclaimers in wrong location.
